dot/core/interface.go

```diff
@@ -42,6 +42,12 @@ type RuntimeInstance interface {
 	DecodeSessionKeys(enc []byte) ([]byte, error)
 	PaymentQueryInfo(ext []byte) (*types.RuntimeDispatchInfo, error)
 	CheckInherents()
+	BabeGenerateKeyOwnershipProof(slot uint64, authorityID [32]byte) (
```

Comment:
Really really like shifting from passing in bytes to actual params and handling from their. This is how substrate does it and is much more intuitive in my opinion

Reply:
```
p_api::decl_runtime_apis! {
	/// API necessary for block authorship with BABE.
	#[api_version(2)]
	pub trait BabeApi {
		...
		/// Generates a proof of key ownership for the given authority in the
		/// current epoch. An example usage of this module is coupled with the
		/// session historical module to prove that a given authority key is
		/// tied to a given staking identity during a specific session. Proofs
		/// of key ownership are necessary for submitting equivocation reports.
		/// NOTE: even though the API takes a `slot` as parameter the current
		/// implementations ignores this parameter and instead relies on this
		/// method being called at the correct block height, i.e. any point at
		/// which the epoch for the given slot is live on-chain. Future
		/// implementations will instead use indexed data through an offchain
		/// worker, not requiring older states to be available.
		fn generate_key_ownership_proof(
			slot: Slot,
			authority_id: AuthorityId,
		) -> Option<OpaqueKeyOwnershipProof>;
		...
	}
}
```
Looks right to me? `AuthorityID` resolves to a sr25519 public key, which is a `[u8; 32]`.